### PR TITLE
Sass feature2

### PR DESCRIPTION
**Pull request by AmandaJisoo**
Prefinal version of the project

### CODE REVIEW

**Comment by kaze-st** [comment]
You need to make sure that the listener is also on the circle when you select the joke type.

**Comment by kaze-st** [comment]
For the phone version, it would be good to have 1 column instead of two. Read about design principles for mobile.

**Comment by kaze-st** [comment]
When I click on the next page button, it should highlight in which page I am at.

**Comment by kaze-st** [comment]
When you expand the size of the screen, make sure that the elements size still are the same. Don't expand it! 😄

**Comment by kaze-st** [comment]
You have a UI bug when the width of the screen is 1187px.

**Comment by kaze-st** [comment]
When you switch jokes, make sure to refresh the page navigation. 💩

**Comment by kaze-st** [comment]
make sure that you can't go to page 6999.

**Comment by kaze-st** [comment]
[Specificity of css rules](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity)

File: index.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          <a class="page-link" id="next-btn" href="#" aria-label="Next">
+            <span aria-hidden="true">&raquo;</span>
+            <span class="sr-only">Next</span>
+          </a>
+        </li>
+      </ul>
+    </nav>
+  </div>
+  <footer>
+    <div class="git-hub-icon icon">
```
Comment (kaze-st) [inline index.html:78]: You should add an aria label here to describe that your icon is a Github icon.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      <a href="https://github.com/AmandaJisoo" id="git-hub"><i class="fal fab fa-github-square fa-3x"></i> </a>                
+    </div>
+    <div class="resume-page icon">
```
Comment (kaze-st) [inline index.html:81]: Same comment as above ^ ❤️

File: index.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      <a href=""><i class="fas fa-user-circle"></i></a>
+   </div> 
+  </footer>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
```
Comment (kaze-st) [inline index.html:85]: Include polyfill fetch or else people with IE won't be able to use `fetch`.

File: scripts.js [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+let state = {curPage: 1, numOfJoke: 10, curJokeLink : null, firstNumOfPageNav: 1}; 
+const ICNDB_URl = "http://api.icndb.com/jokes/";
+const DADDY_URL = "https://icanhazdadjoke.com/";
+
+let curPage = 1; 
+let numOfJoke = 10;
+let curJokeLink = null;
+let firstNumOfPageNav = 1;
+let isAtHome = true;
```
Comment (kaze-st) [inline scripts.js:11]: Make sure to refractor the variables into the state.

File: scripts.js [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+$("#icndb").click(renderHomePage);
+$("#daddy").click(renderHomePage);
+$("#assport").click(renderHomePage);
+$("#next-btn").click(changeToNextPageNav);
+$("#prev-btn").click(changeToPrevPageNav);
+$(".page-btn").click(loadNextPage);
+$("#project-description").click(displayProjectInfo);
+$(".paper").hide();
+$("#home").click(resetToHome);
+
```
Comment (kaze-st) [inline scripts.js:23]: You are assigning listeners here. You should have a function, i.e `initListeners()`, that encapsulates this and call it afterwards.

File: scripts.js [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+        return values.map((response) => response.json())
+    }).then((response =>{
+        Promise.all(response).then(icndbAppendToPage);
+    }))
+    .then(()=> {
+        $("#loading").hide();
+        $("footer").show();
+        $(".pagination").show();
+    })
+    .catch(console.error);
```
Comment (kaze-st) [inline scripts.js:189]: You need to handle the error gracefully. Console logging it out is not enough. For instance, you can display a modal in the page in case there is an error.

File: scripts.js [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+
+/**
+ *  function is called when user click one 
+ *  of the joke option
+ *  hides items items displayed on the home page
+ *  identifies the joke the user clicked
+ * @param {event} returns the event triggered by
+ * the option user chose on the form
+ */
+function renderHomePage(event) {
```
Comment (kaze-st) [inline scripts.js:71]:
```
   function is called when user click one 
 of the joke option
  hides items items displayed on the home page
  identifies the joke the user clicked
```

The description of your function does not follow the actual naming of your function. Do something like handleJokeClick? RenderJokes? 

Also you are not returning anything, why document it 😢

File: scripts.js [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+        curJokeLink = "daddy";
+        daddyFetch();
+    } 
+}
+
+/**
+ * this function identifies the type of joke
+ * that user selected and make appropriate fetch call
+ */
+function checkJokeTypeForFetch() {
```
Comment (kaze-st) [inline scripts.js:91]: Functions that starts with the word `check` is usually a bad naming. You description here is misleading as well. What this function actually do is to fetch a set of jokes from the designated page.

File: scripts.js [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+/**
+ * this function is called when user clicks the backward
+ * btn of the pagination
+ * hides the currently displayed items 
+ * and generate the correct pagination  for upcoming display
+ * Display loading icons to alarm user that their request has been received
+ * @param {event} triggered when user click the backward btn
+ * this function doesn't respond on the first pagination display(pg 1, 2,3)
+ */
+function changeToPrevPageNav(event) {
```
Comment (kaze-st) [inline scripts.js:147]: A better name could be `hanldeChangeToPrevPage`. Same goes for the for the function above ^

File: scripts.js [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+    .then(checkStatus)
+    .then(resp => resp.json())
+    .then((response) => {
+        daddyAppendToPage(response);
+    }).then(()=> {
+        $("#loading").hide();
+        $(".pagination").show();
+        $("footer").show();
+    })
+    .catch(console.error);
```
Comment (kaze-st) [inline scripts.js:235]: 👎

File: scss/style.scss [hunk 11]
```diff
@@ -0,0 +1,420 @@
...
+        cursor: pointer;
+    }
+    
+    ul li:hover label {
+        color: #FFFFFF;
+    }
+    
+    ul li .check {
+        display: block;
+        position: absolute;
```
Comment (kaze-st) [inline scss/style.scss:175]: Position absolute gives me PTSD. Please use another position 👎

File: scss/style.scss [hunk 12]
```diff
@@ -0,0 +1,420 @@
...
+        color: black;
+        display: block;
+        position: relative;
+        float: left;
+        height: 100px;
+        border-bottom: 1px solid #333;
+    }
+    
+    ul li input[type=radio] {
+        position: absolute;
```
Comment (kaze-st) [inline scss/style.scss:155]: 🥇

File: scss/style.scss [hunk 13]
```diff
@@ -0,0 +1,420 @@
...
+
+    .fa-user-circle:before {
+        content: "\f2bd";
+        font-size: 105%;
+        margin-right: 11px;
+    }
+}
+
+
+@media (min-width: 600px) and (max-width : 1024px) {
```
Comment (kaze-st) [inline scss/style.scss:288]: Don't have max width.

**Review by kaze-st — CHANGES_REQUESTED** [review]

**Comment by AmandaJisoo** [comment]
> For the phone version, it would be good to have 1 column instead of two. Read about design principles for mobile.

Resolved ><
